### PR TITLE
LEAF-1441 fortify report

### PR DESCRIPTION
**Pull request by patrickcgs**
Updates pipeline to generate fortify report

### CODE REVIEW

File: fort_report/FortifyReportGen.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#cd /workspace/source_code/
+#ls
+#ls /workspace/fort_report/fortify.properties
+
+fortifyupdate
+
+echo --------------------------------------
+echo Cleaning previous scan artifacts...
+sourceanalyzer $MEMORY -b $ARTIFACT_ID -build-label $ARTIFACT_ID -clean -verbose
+
```
Comment (dogwang94) [inline fort_report/FortifyReportGen.sh:42]: should we have

if [ $? -ne 0 ]; then
echo “Sourceanalzyer failed, exiting”
exit 1
fi

File: fort_report/FortifyReportGen.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+echo --------------------------------------
+echo Translating project...
+sourceanalyzer -php-source-root libs/ $MEMORY $LAUNCHERSWITCHES -b $ARTIFACT_ID -build-label $ARTIFACT_ID @$PROPERTIES_FILE
+
```
Comment (dogwang94) [inline fort_report/FortifyReportGen.sh:46]: should we give a message if scan is failed?


if [ $? -ne 0 ]; then
echo “Sourceanalzyer failed, exiting”
exit 1
fi

File: fort_report/FortifyReportGen.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+echo --------------------------------------
+echo Starting scan
+sourceanalyzer $MEMORY $LAUNCHERSWITCHES -b $ARTIFACT_ID -build-label $ARTIFACT_ID -scan -f $FPR -verbose
+
```
Comment (dogwang94) [inline fort_report/FortifyReportGen.sh:50]: should we give a message if scan is failed?

if [ $? -ne 0 ]; then
echo “Sourceanalzyer failed, exiting”
exit 1
fi

File: fort_report/FortifyReportGen.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+MEMORY="-Xmx1600M -Xms1000M -Xss48M"
+
+TEMPLATE="/workspace/fortify_templates/Security_Report.xml"
+REPORT_OPTIONS="-showRemoved -showSuppressed -showHidden -verbose"
+
+set -x
+# enable debug
+
+set -eo pipefail
```
Comment (patrickcgs) [inline fort_report/FortifyReportGen.sh:30]: Script exists and fails pipeline if commands below return non-zero status code.

File: fort_report/FortifyReportGen.sh (hunk 1 above)
Comment (patrickcgs) [reply fort_report/FortifyReportGen.sh:40]: Good catch.  Yes, it is a good idea to have the script exit if any error is encountered.  I choose to implement it instead with bash options:

https://github.com/VHAINNOVATIONS/LEAF/pull/962/files#diff-a3cc50141014cce8be14df326ff9d01dR27-R30

I also added a check in the Jenkinsfile to ensure that the artifact produced from the script is present:

https://github.com/VHAINNOVATIONS/LEAF/pull/962/files#diff-37174e57f1913929c66bc843aca864d6R50

File: fort_report/FortifyReportGen.sh (hunk 2 above)
Comment (patrickcgs) [reply fort_report/FortifyReportGen.sh:44]: See comment above

File: fort_report/FortifyReportGen.sh (hunk 3 above)
Comment (patrickcgs) [reply fort_report/FortifyReportGen.sh:48]: See comment above

**Comment by thomasnguyen95** [comment]
Are there any SQL related errors from running the scan script that was coming up from manual scans?

**Comment by patrickcgs** [comment]
> @patrickcgs Are there any SQL related errors from running the scan script that was coming up from manual scans?

If so, they will be addressed in separate story for resolving scan errors.

**Pull request merged**